### PR TITLE
Prohibited matches in filters and outputs in telemetry-operator

### PR DESCRIPTION
**Pull request by Teneroy**
**Description**

Changes proposed in this pull request:

- Prohibited "Match" in filters and outputs
- Fixed validator tests
- Refactored tests to adapt to the new logic
- Changed examples

**Related issue(s)**
#14562

### CODE REVIEW

**Review by NHingerl — DISMISSED** [review]

File: components/telemetry-operator/internal/validation/plugin_validator_test.go [hunk 1]
```diff
@@ -39,6 +39,7 @@ func TestValidateForbiddenFilters(t *testing.T) {
 	err := sut.Validate(logPipeline, logPipelines)
 
 	require.Error(t, err)
+	assert.Contains(t, err.Error(), "plugin 'grep' contains match condition. Match conditions are forbidden")
```
Comment (chrkl) [inline components/telemetry-operator/internal/validation/plugin_validator_test.go:42]: Looks like the test before did not what the name suggests. Let's split it to two tests. One to validate that a LogPipeline that has a `match` expression is rejected and one that tests for the forbidden `lua` filter.

File: components/telemetry-operator/internal/validation/plugin_validator_test.go [hunk 2]
```diff
@@ -57,14 +58,15 @@ func TestValidateForbiddenOutputs(t *testing.T) {
 	err := sut.Validate(logPipeline, logPipelines)
 
 	require.Error(t, err)
+	assert.Contains(t, err.Error(), "plugin 'es' contains match condition. Match conditions are forbidden")
```
Comment (chrkl) [inline components/telemetry-operator/internal/validation/plugin_validator_test.go:61]: Same here, let's also test that the `es` output is forbidden.

**Review by chrkl — CHANGES_REQUESTED** [review]

**Review by chrkl — APPROVED** [review]

**Pull request merged**